lib/streamDAO.js

```diff
@@ -195,6 +195,19 @@ module.exports = function(mongoClient){
           }
         });
       });
+    },
+    // This function is used to delete existing CareLink data from an account
+    // before uploading new -- but only that data that doesn't have an uploadId
+    deleteOldCarelinkData: function(groupId, cb) {
+      mongoClient.withCollection('deviceData', cb, function(collection, cb) {
+        collection.remove({ groupId: groupId, source: 'carelink', uploadId: {$exists: false} }, function(err){
+          if (err != null) {
+            cb(err);
+          } else {
+            collection.remove({ _groupId: groupId, source: 'carelink', uploadId: {$exists: false} }, cb);
```

Comment:
Is this here a second time with `_groupId` because some very old accounts have that field name instead of `groupId`?